php-frontend/src/main/java/org/sonar/php/parser/LexicalConstant.java

```diff
@@ -65,7 +65,7 @@
    */
   private static final String SINGLE_LINE_COMMENT_CONTENT = "(?:(?!" + PHP_CLOSING_TAG + ")[^\\n\\r])*+";
   private static final String SINGLE_LINE_COMMENT1 = "//" + SINGLE_LINE_COMMENT_CONTENT;
-  private static final String SINGLE_LINE_COMMENT2 = "#" + SINGLE_LINE_COMMENT_CONTENT;
+  private static final String SINGLE_LINE_COMMENT2 = "#[^\\[]" + SINGLE_LINE_COMMENT_CONTENT;
```

Comment:
I think that doesn't work for empty comments with just "#". In such cases the following character is consumed: it's usually a line end and the next line is then wrongly part of the comment:
```
<?php
#
$a
 = 42;
```
A parsing error is raised when there's a single # at the very end of the file:
```
<?php
#
```